06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js

```diff
@@ -0,0 +1,221 @@
+/* global FileReader */
+/* global InvertedIndex */
+/**
+ * @class InvertedIndex
+ * @classdesc blah blah
+ */
+/* eslint-disable */
+class InvertedIndex {
+  /* eslint-enable */
+  /**
+   * * @constructor
+   * initialises the class base properties
+   */
+  constructor() {
+    this.indicies = {};
+    this.indexedFiles = {};
+    this.uploadedFiles = {};
+  }
+  /**
+   * @createIndex method
+   * @param {fileName} fileName
+   * @param {fileContent} fileContent
+   * @returns {boolean}
+   * create index of the fileName
+   */
+  createIndex(fileName, fileContent) {
+    this.indicies[fileName] = this.indicies[fileName] || {};
+    const numOfBooks = fileContent.length;
+    for (let bookIndex = 0; bookIndex <
+      numOfBooks; bookIndex += 1) {
+      const { title, text } = fileContent[bookIndex],
+        tokens = InvertedIndex.tokenize(`${title} ${text}`),
+        indicies = this.indicies[fileName];
+      tokens.forEach((token) => {
+        if (token in indicies) {
+          const eachToken = indicies[token];
+          if (eachToken.indexOf(bookIndex) === -1) {
+            indicies[token].push(bookIndex);
+          }
+        } else {
+          // Initially this is what happens
+          indicies[token] = [bookIndex];
+        }
+      });
+    }
+
+    this.indexedFiles[fileName] = numOfBooks;
+    return true;
+  }
+
+  /**
+   * @getIndex method
+   * @param {fileName} fileName
+   * @returns {Object}
+   * gets the index of the fileName
+   */
+  getIndex(fileName) {
+    return this.indicies[fileName];
+  }
+
+   /**
+   * @getNumofBooks method
+   * @param {fileName} fileName
+   * @returns {Array}
+   * create index of the fileName
+   */
+  getNumOfBooks(fileName) {
```

Comment:
Please be descriptive with your naming. Abbreviations are not allowed `getNumOfBooks`